In [1]:
import os
SNAPPY_notebook_path = os.path.join(os.path.abspath(""), "bench.ipynb")

In [2]:
import time
SNAPPY_start_time = time.perf_counter_ns()

In [3]:
%load_ext autotime

time: 51.3 µs (started: 2024-05-07 14:57:17 -04:00)


In [4]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import os

import pandas as pd


# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

time: 234 ms (started: 2024-05-07 14:57:17 -04:00)


In [5]:
data = pd.read_csv(os.path.abspath('') + "/input/Billionaires Statistics Dataset.csv", index_col="rank")

time: 20.7 ms (started: 2024-05-07 14:57:17 -04:00)


In [6]:
factor = 800
data = pd.concat([data]*factor, ignore_index=True)
# data.info()

time: 185 ms (started: 2024-05-07 14:57:17 -04:00)


In [7]:
data.head(5)

finalWorth               category                personName   age  \
0      211000       Fashion & Retail  Bernard Arnault & family  74.0   
1      180000             Automotive                 Elon Musk  51.0   
2      114000             Technology                Jeff Bezos  59.0   
3      107000             Technology             Larry Ellison  78.0   
4      106000  Finance & Investments            Warren Buffett  92.0   

         country    city              source             industries  \
0         France   Paris                LVMH       Fashion & Retail   
1  United States  Austin       Tesla, SpaceX             Automotive   
2  United States  Medina              Amazon             Technology   
3  United States   Lanai              Oracle             Technology   
4  United States   Omaha  Berkshire Hathaway  Finance & Investments   

  countryOfCitizenship                      organization  ...  \
0               France  LVMH Moët Hennessy Louis Vuitton  ...   
1        United States                             Tesla  ...   
2        United States                            Amazon  ...   
3        United States                            Oracle  ...   
4        United States    Berkshire Hathaway Inc. (Cl A)  ...   

   cpi_change_country           gdp_country  \
0                 1.1   $2,715,518,274,227    
1                 7.5  $21,427,700,000,000    
2                 7.5  $21,427,700,000,000    
3                 7.5  $21,427,700,000,000    
4                 7.5  $21,427,700,000,000    

  gross_tertiary_education_enrollment  \
0                                65.6   
1                                88.2   
2                                88.2   
3                                88.2   
4                                88.2   

  gross_primary_education_enrollment_country life_expectancy_country  \
0                                      102.5                    82.5   
1                                      101.8                    78.5   
2                                      101.8                    78.5   
3                                      101.8                    78.5   
4                                      101.8                    78.5   

  tax_revenue_country_country total_tax_rate_country population_country  \
0                        24.2                   60.7         67059887.0   
1                         9.6                   36.6        328239523.0   
2                         9.6                   36.6        328239523.0   
3                         9.6                   36.6        328239523.0   
4                         9.6                   36.6        328239523.0   

  latitude_country longitude_country  
0        46.227638          2.213749  
1        37.090240        -95.712891  
2        37.090240        -95.712891  
3        37.090240        -95.712891  
4        37.090240        -95.712891  

[5 rows x 34 columns]

time: 12.8 ms (started: 2024-05-07 14:57:18 -04:00)


In [8]:
# data.describe()

time: 8.31 ms (started: 2024-05-07 14:57:18 -04:00)


In [9]:
country_names =data["country"].value_counts() #List of how many billionaires there are in the country

time: 58.6 ms (started: 2024-05-07 14:57:18 -04:00)


In [10]:
country_names 

country
United States           603200
China                   418400
India                   125600
Germany                  81600
United Kingdom           65600
                         ...  
Portugal                   800
Georgia                    800
Eswatini (Swaziland)       800
Uzbekistan                 800
Armenia                    800
Name: count, Length: 78, dtype: int64

time: 11 ms (started: 2024-05-07 14:57:18 -04:00)


> > > > >  ****10 countries with the most billionaires****

In [11]:
data_100 = data.loc[:100,["finalWorth","category","country"]]


time: 25.4 ms (started: 2024-05-07 14:57:18 -04:00)


In [12]:
data_100_category = data_100["category"].value_counts()

time: 580 µs (started: 2024-05-07 14:57:18 -04:00)


> > > > > ****Most Richest 100 person in the world and their categories****

In [13]:
data_usa = data[data["country"]== "United States"] #We focus on billionaires based in United States


time: 158 ms (started: 2024-05-07 14:57:18 -04:00)


In [14]:
data_usa_category = data_usa["category"].value_counts()
data_usa_category

category
Finance & Investments         152000
Technology                    112800
Food & Beverage                59200
Fashion & Retail               50400
Real Estate                    40800
Media & Entertainment          29600
Energy                         28000
Sports                         28000
Healthcare                     27200
Manufacturing                  22400
Service                        15200
Diversified                    10400
Automotive                      9600
Gambling & Casinos              4800
Telecom                         4000
Construction & Engineering      4000
Logistics                       4000
Metals & Mining                  800
Name: count, dtype: int64

time: 16.2 ms (started: 2024-05-07 14:57:18 -04:00)


> > > > > ****All Billionaires in the USA and their categories****

In [15]:
data_usa_city = data_usa["city"].value_counts()
print(data_usa_city)
# data_usa_city.info()

city
New York             79200
San Francisco        29600
Los Angeles          27200
Palm Beach           16800
Dallas               16000
                     ...  
Steamboat Springs      800
Westlake               800
Coconut Grove          800
Short Hills            800
Pottsville             800
Name: count, Length: 268, dtype: int64
time: 24.8 ms (started: 2024-05-07 14:57:18 -04:00)


268 city exist, I'll try .head(20)

> > > > > **The 20 cities has most billionaires in usa**

In [16]:
data.head(5)

finalWorth               category                personName   age  \
0      211000       Fashion & Retail  Bernard Arnault & family  74.0   
1      180000             Automotive                 Elon Musk  51.0   
2      114000             Technology                Jeff Bezos  59.0   
3      107000             Technology             Larry Ellison  78.0   
4      106000  Finance & Investments            Warren Buffett  92.0   

         country    city              source             industries  \
0         France   Paris                LVMH       Fashion & Retail   
1  United States  Austin       Tesla, SpaceX             Automotive   
2  United States  Medina              Amazon             Technology   
3  United States   Lanai              Oracle             Technology   
4  United States   Omaha  Berkshire Hathaway  Finance & Investments   

  countryOfCitizenship                      organization  ...  \
0               France  LVMH Moët Hennessy Louis Vuitton  ...   
1        United States                             Tesla  ...   
2        United States                            Amazon  ...   
3        United States                            Oracle  ...   
4        United States    Berkshire Hathaway Inc. (Cl A)  ...   

   cpi_change_country           gdp_country  \
0                 1.1   $2,715,518,274,227    
1                 7.5  $21,427,700,000,000    
2                 7.5  $21,427,700,000,000    
3                 7.5  $21,427,700,000,000    
4                 7.5  $21,427,700,000,000    

  gross_tertiary_education_enrollment  \
0                                65.6   
1                                88.2   
2                                88.2   
3                                88.2   
4                                88.2   

  gross_primary_education_enrollment_country life_expectancy_country  \
0                                      102.5                    82.5   
1                                      101.8                    78.5   
2                                      101.8                    78.5   
3                                      101.8                    78.5   
4                                      101.8                    78.5   

  tax_revenue_country_country total_tax_rate_country population_country  \
0                        24.2                   60.7         67059887.0   
1                         9.6                   36.6        328239523.0   
2                         9.6                   36.6        328239523.0   
3                         9.6                   36.6        328239523.0   
4                         9.6                   36.6        328239523.0   

  latitude_country longitude_country  
0        46.227638          2.213749  
1        37.090240        -95.712891  
2        37.090240        -95.712891  
3        37.090240        -95.712891  
4        37.090240        -95.712891  

[5 rows x 34 columns]

time: 20.1 ms (started: 2024-05-07 14:57:18 -04:00)


In [17]:
category_list = list(data.category.unique())
category_list

['Fashion & Retail',
 'Automotive',
 'Technology',
 'Finance & Investments',
 'Media & Entertainment',
 'Telecom',
 'Diversified',
 'Food & Beverage',
 'Logistics',
 'Gambling & Casinos',
 'Manufacturing',
 'Real Estate',
 'Metals & Mining',
 'Energy',
 'Healthcare',
 'Service',
 'Construction & Engineering',
 'Sports']

time: 65.7 ms (started: 2024-05-07 14:57:18 -04:00)


In [18]:
data.finalWorth = data.finalWorth.astype(float)
worth_average = []
for i in category_list:
    x = data[data['category']==i]
    worth_ratio = sum(x.finalWorth)/len(x)
    worth_average.append(worth_ratio)
data2 = pd.DataFrame({'category_list': category_list,'worth_average':worth_average})
new_index = (data2['worth_average'].sort_values(ascending=False)).index.values
new_data = data2.reindex(new_index)

time: 2.01 s (started: 2024-05-07 14:57:18 -04:00)


In [19]:
data3 =data.dropna(axis="index",how="any",inplace=False) # ı created new df because so many nan values was exist

time: 800 ms (started: 2024-05-07 14:57:20 -04:00)


In [20]:
data3.head() 

finalWorth               category      personName   age        country  \
1    180000.0             Automotive       Elon Musk  51.0  United States   
2    114000.0             Technology      Jeff Bezos  59.0  United States   
3    107000.0             Technology   Larry Ellison  78.0  United States   
4    106000.0  Finance & Investments  Warren Buffett  92.0  United States   
5    104000.0             Technology      Bill Gates  67.0  United States   

     city              source             industries countryOfCitizenship  \
1  Austin       Tesla, SpaceX             Automotive        United States   
2  Medina              Amazon             Technology        United States   
3   Lanai              Oracle             Technology        United States   
4   Omaha  Berkshire Hathaway  Finance & Investments        United States   
5  Medina           Microsoft             Technology        United States   

                      organization  ...  cpi_change_country  \
1                            Tesla  ...                 7.5   
2                           Amazon  ...                 7.5   
3                           Oracle  ...                 7.5   
4   Berkshire Hathaway Inc. (Cl A)  ...                 7.5   
5  Bill & Melinda Gates Foundation  ...                 7.5   

            gdp_country gross_tertiary_education_enrollment  \
1  $21,427,700,000,000                                 88.2   
2  $21,427,700,000,000                                 88.2   
3  $21,427,700,000,000                                 88.2   
4  $21,427,700,000,000                                 88.2   
5  $21,427,700,000,000                                 88.2   

  gross_primary_education_enrollment_country life_expectancy_country  \
1                                      101.8                    78.5   
2                                      101.8                    78.5   
3                                      101.8                    78.5   
4                                      101.8                    78.5   
5                                      101.8                    78.5   

  tax_revenue_country_country total_tax_rate_country population_country  \
1                         9.6                   36.6        328239523.0   
2                         9.6                   36.6        328239523.0   
3                         9.6                   36.6        328239523.0   
4                         9.6                   36.6        328239523.0   
5                         9.6                   36.6        328239523.0   

  latitude_country longitude_country  
1         37.09024        -95.712891  
2         37.09024        -95.712891  
3         37.09024        -95.712891  
4         37.09024        -95.712891  
5         37.09024        -95.712891  

[5 rows x 34 columns]

time: 10.8 ms (started: 2024-05-07 14:57:21 -04:00)


In [21]:
# data3.info() #in the new df we have 238 rows for each column

time: 8.59 ms (started: 2024-05-07 14:57:21 -04:00)


In [22]:
data3.country.unique()  # and 238 billionaires in usa, others deleted.

array(['United States'], dtype=object)

time: 18.3 ms (started: 2024-05-07 14:57:21 -04:00)


In [23]:
SNAPPY_end_time = time.perf_counter_ns()
print("Total elapsed time:", (SNAPPY_end_time - SNAPPY_start_time) / (10 ** 9))

Total elapsed time: 3.785565175
time: 3.03 ms (started: 2024-05-07 14:57:21 -04:00)


**kernel will continue...**